src/evaluator.ts

```
@@ -45,19 +46,21 @@ export class Evaluator {
   }
 
   evaluateFlag(
+    configuration: Configuration,
     flag: Flag,
-    configDetails: ConfigDetails,
     subjectKey: string,
     subjectAttributes: Attributes,
-    obfuscated: boolean,
     expectedVariationType?: VariationType,
   ): FlagEvaluation {
+    const flagsConfig = configuration.getFlagsConfiguration();
     const flagEvaluationDetailsBuilder = new FlagEvaluationDetailsBuilder(
-      configDetails.configEnvironment.name,
+      flagsConfig?.response.environment.name ?? '',
       flag.allocations,
-      configDetails.configFetchedAt,
-      configDetails.configPublishedAt,
+      flagsConfig?.fetchedAt ?? '',
+      flagsConfig?.response.createdAt ?? '',
     );
+    const configFormat = flagsConfig?.response.format;
+    const obfuscated = configFormat !== FormatEnum.SERVER;
```

Should `isObfuscated` be a method on `Configuration` so we're not leaking the FormatEnum details?

---

`isObfuscated()` is oversimplification in a sense that `Configuration` contains multiple pieces (flags, bandits, precomputed), where each could be obfuscated, not obfuscated, or simply missing. So a single `isObfuscated()` method doesn't cut it.

It also spreads this oversimplified model in programmers' brains, so we're more likely to miss some edge cases. I would rather have us working with raw data instead of half-baked abstractions. It may not be as pretty (initially) but at least it's obvious what's going on

re leaking details: my philosophy is that SDKs are in the know. Evaluation is _highly_ dependent on flags response type, so there's no sense hiding it. When talking about leaking details, the only meaningful distinction for me is user–SDK. `Configuration` is exposing full server responses but is marking them `@internal` for this reason

---

re `FormatEnum` specifically: it's currently mixing formats from two independent and unrelated configuration pieces (flags and precomputed), that actually have different rules for determining obfuscation

Next refactoring step is splitting it into `FlagsConfigurationFormat` and `PrecomputedConfigurationFormat`. And then we can have a proper `isObfuscated(FlagsConfigurationFormat): boolean` helper to encode this piece of knowledge

Longer-term still, we shall add `isObfuscated` field to flags configuration

---

I don't think we need to handle differently formatted flags and bandits and whatnot. If bandit format doesn't match flags we could just throw an error or something.